### PR TITLE
chore(v5): update all docs, selectors, css vars, scss vars to include v5

### PR DESCRIPTION
**Pull request by mattnolting**
related to patternfly/patternfly#5376 

This PR:

- `pf-{c,l,u}-` to `pf-v5-{c,l,u}`, `pf-global` to `${#pf-global}`
- `pf-global` -> `${#pf-global}`
- updates all sass references to `pf-v5-...`
- adds `{name[c,l,u]}` as a sass variable for each component/layout/utility which is the called as `#{$name[c,l,u]`
- updates all documentation
- adds a handlebars helper that inserts the prefix in `.hbs` templates
- adds a global variable for version `$version-root`, `$pf-prefix`, `$pf-global` (intentionally unversioned as it's referenced elsewhere), ` in https://github.com/patternfly/patternfly/pull/5434/files#diff-ae9b224fabde000b6b908e9a4bfe412436d3fc19d65c5b4249f344f23a7d88ab
```
$version-root: 5;

// Patternfly options
$pf-v5-global--namespace: 'pf-' !default;
$pf-v5-global--version: 'v#{$version-root}-' !default;

// Namespace prefix and version
$pf-prefix: #{$pf-v5-global--namespace} + #{$pf-v5-global--version};

// Global varilble, namespace prefix and version
$pf-global: #{$pf-prefix} + '-global';
```

### CODE REVIEW

File: src/patternfly/sass-utilities/scss-variables.scss [hunk 1]
```diff
@@ -5,6 +5,9 @@
 // `$pf-global--concept--PropiertyCamelCase--modifier--state`
 
 // Patternfly options
+$pf-global--prefix: 'pf';
+$pf-global--version: 'v5';
+$pf-global-var: '--' + #{$pf-global--prefix} + '-' + #{$pf-global--version} + '-global--';
```
Comment (mcoker) [inline src/patternfly/sass-utilities/scss-variables.scss:10]: should you add !default so these can be overridden?

**Comment by mattnolting** [comment]
@mcoker @srambach @dgutride I believe this is ready for review and testing!

File: src/patternfly/utilities/Sizing/sizing.scss [hunk 2]
```diff
@@ -41,49 +41,49 @@ $pf-u-height-options: (
...
 );
 
-@include pf-utility-builder($pf-u-width-options, $pf-global--breakpoint-list);
-@include pf-utility-builder($pf-u-height-options, $pf-global--breakpoint-list);
+@include pf-v5-utility-builder($pf-v5-u-width-options, $pf-v5-global--breakpoint-list);
+@include pf-v5-utility-builder($pf-v5-u-height-options, $pf-v5-global--breakpoint-list);
 
-.pf-u-min-width {
-  --pf-u-min-width--MinWidth: 0;
+.#{$pf-global}-u-min-width {
```
Comment (mcoker) [inline src/patternfly/utilities/Sizing/sizing.scss:47]: Looks like  most/all utility classnames are broken - I think you meant to use `$pf-prefix` here

File: src/patternfly/components/Card/card.scss [hunk 3]
```diff
@@ -1,150 +1,152 @@
...
 
 
   // Selectable/Clickable
-  --pf-c-card--m-selectable--BorderWidth: var(--pf-global--BorderWidth--sm);
+  --#{$card}--m-selectable--BorderWidth: var(--pf-global--BorderWidth--sm);
 
   // Selected state (checked) of selectable card
-  --pf-c-card--m-selectable--m-selected--BackgroundColor: var(--pf-global--BackgroundColor--200);
-  --pf-c-card--m-selectable--m-selected--BorderColor: var(--pf-global--active-color--100);
+  --#{$card}--m-selectable--m-selected--BackgroundColor: var(--pf-global--BackgroundColor--200);
```
Comment (mcoker) [inline src/patternfly/components/Card/card.scss:36]: Looks like there are a handful of vars that need to reference `$pf-global` in this stylesheet

```suggestion
  --#{$card}--m-selectable--m-selected--BackgroundColor: var(--#{$pf-global}--BackgroundColor--200);
```

**Review by mcoker — COMMENTED** [review]
Just reviewed the basic setup and noted a few things. Overall looks good to me. I'd probably opt to use `#{$pf-prefix}-c-[component-name]` instead of creating unique vars per component/layout just to keep that part consistent across the board (we're replacing any instance of the "pf" prefix with "pf-v5" and leaving the rest), and allows us to search by 'l-flex--Var' and 'u-flex--Var', but I also like less repetition, so I'll defer to whatever yours/Sarah's preference is.

Also I noticed some scss isn't compiling, looks like you might need to make this kinda change:

```diff
diff --git a/src/patternfly/base/_chart-globals.scss b/src/patternfly/base/_chart-globals.scss
index 66f67786..678f0706 100644
--- a/src/patternfly/base/_chart-globals.scss
+++ b/src/patternfly/base/_chart-globals.scss
@@ -1,3 +1,6 @@
+@import "../sass-utilities/version";
+@import "../sass-utilities/prefixes";
+@import "../sass-utilities/component-namespaces";
 @import "../sass-utilities/colors";
 @import "../sass-utilities/scss-variables";
 
diff --git a/src/patternfly/base/themes/dark/_chart-globals.scss b/src/patternfly/base/themes/dark/_chart-globals.scss
index b2b1c284..346f9ed3 100644
--- a/src/patternfly/base/themes/dark/_chart-globals.scss
+++ b/src/patternfly/base/themes/dark/_chart-globals.scss
@@ -1,3 +1,6 @@
+@import "../../../sass-utilities/version";
+@import "../../../sass-utilities/prefixes";
+@import "../../../sass-utilities/component-namespaces";
 @import "../../../sass-utilities/colors";
 @import "../../../sass-utilities/scss-variables";
 
diff --git a/src/patternfly/patternfly-date-picker.scss b/src/patternfly/patternfly-date-picker.scss
index 020ed22d..d2105b89 100644
--- a/src/patternfly/patternfly-date-picker.scss
+++ b/src/patternfly/patternfly-date-picker.scss
@@ -1,3 +1,7 @@
+@import "./sass-utilities/version";
+@import "./sass-utilities/prefixes";
+@import "./sass-utilities/component-namespaces";
```

Re the above:
1. These new things could go in an `init.scss` file or something like that, so they can be included together
2. You may just be able to change those imports to `sass-utilities/all`, though I'd just doublecheck that the `placeholders.scss` file doesn't get rendered in the compiled CSS for those files.
3. Can we just remove [patternfly-date-picker.scss](https://github.com/patternfly/patternfly/blob/v5/src/patternfly/patternfly-date-picker.scss)? @nicolethoen @tlabaj @mcarrano @andrew-ronaldson @mceledonia  those were some styles we added for a flatpickr calendar back in September of 2020. Are those still needed for any reason you're aware of?

**Comment by tlabaj** [comment]
> 3\. Can we just remove [patternfly-date-picker.scss](https://github.com/patternfly/patternfly/blob/v5/src/patternfly/patternfly-date-picker.scss)? @nicolethoen @tlabaj @mcarrano @andrew-ronaldson @mceledonia  those were some styles we added for a flatpickr calendar back in September of 2020. Are those still needed for any reason you're aware of?

@mcoker that is not being used in react so I believe it is safe to remove.

File: scripts/template/src/patternfly/{folder}/{name}/{nameDasherized}.scss [hunk 4]
```diff
@@ -1,26 +1,27 @@
 // Component variables follow this formula:
-// --{nameBEM}__element--modifier--state--PropertyCamelCase
+// --#{${nameBEM}}__element--modifier--state--PropertyCamelCase
+// @debug $about-modal-box; // check your variable names located in src/patternfly/sass-utilities/component-namespaces
 
 // The value of component scoped variables is always defined by a global variable:
-.{nameBEM} {
+.#{$#{${nameBEM}}} {
```
[the remark this answers is not in the document]
Comment (mattnolting) [reply scripts/template/src/patternfly/{folder}/{name}/{nameDasherized}.scss]: I'll need to retest after `node generate` update

File: patternfly-docs/site/pages/guidelines.md [hunk 5]
```diff
@@ -48,7 +48,7 @@ PatternFly is made up of isolated components that don't allow dependencies. Ther
 
 However, from time to time it is recognized that an exception to the PatternFly styling may be needed for a special case. For those instances, utility classes are supplied to assist in allowing minor styling changes without creating the need for adding custom CSS.
 
-A utility class is prefixed with `-u` (after the PatternFly prefix `pf-`), for example: `.pf-u-align-content-center`.
+A utility class is prefixed with `-u` (after the PatternFly prefix `pf-`), for example: `.pf-v5-u-align-content-center`.
```
[the remark this answers is not in the document]
Comment (mattnolting) [reply patternfly-docs/site/pages/guidelines.md:51]: @mcoker https://github.com/patternfly/patternfly/issues/5471

File: scripts/template/src/patternfly/{folder}/{name}/{nameDasherized}.scss (hunk 4 above)
[the remark this answers is not in the document]
Comment (mcoker) [reply scripts/template/src/patternfly/{folder}/{name}/{nameDasherized}.scss]: @mattnolting #5465 was merged, so you should be good to test now 👍

**Comment by tlabaj** [comment]
@mnolting @mcoker this may not be relevant to this PR,  but I just thought about it since you asked the question about [patternfly-date-picker.scss](https://github.com/patternfly/patternfly/blob/v5/src/patternfly/patternfly-date-picker.scss)

This [react PR](https://github.com/patternfly/patternfly-react/pull/8864/files) removes the only use of  'data-list-grid.scss' from react.  Is it still needed?

**Comment by srambach** [comment]
> Just reviewed the basic setup and noted a few things. Overall looks good to me. I'd probably opt to use `#{$pf-prefix}-c-[component-name]` instead of creating unique vars per component/layout just to keep that part consistent across the board (we're replacing any instance of the "pf" prefix with "pf-v5" and leaving the rest), and allows us to search by 'l-flex--Var' and 'u-flex--Var', but I also like less repetition, so I'll defer to whatever yours/Sarah's preference is.

I don't think this bothers me - I like having less repetition.

A few questions:
- There are a lot of instances of "pf-v5" in various scss variables throughout the code. Is the expectation that we will search and replace these for each major release?
- Do mixins need to be versioned? 
- I find a bunch of `pf-v5` in some .hbs - can those use `{{pfv}}`?
- +1 to finding a shortcut for the .md files to include the version

**Comment by mattnolting** [comment]
> @mnolting @mcoker this may not be relevant to this PR, but I just thought about it since you asked the question about [patternfly-date-picker.scss](https://github.com/patternfly/patternfly/blob/v5/src/patternfly/patternfly-date-picker.scss)
> 
> This [react PR](https://github.com/patternfly/patternfly-react/pull/8864/files) removes the only use of 'data-list-grid.scss' from react. Is it still needed?

@tlabaj `data-list-grid.scss` is imported into `data-list.scss`, so if `-grid.scss` is still available for `data-list.scss` to compile, then you can remove it from the React component.

**Review by dgutride — APPROVED** [review]

**Pull request merged**